crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1890,35 +1899,45 @@ impl<'src> Parser<'src> {
 
     /// Parses a `yield from` expression.
     ///
+    /// This method should not be used directly. Use [`Parser::parse_yield_expression`]
+    /// even when parsing a `yield from` expression.
+    ///
```

Comment:
Nit: Maybe rename to `parse_yield_from_expression_impl` to make it somewaht more explicit.